### PR TITLE
fix: don't explode for non-tuple argument values.

### PR DESCRIPTION
**Pull request by jimsynz**
Not sure how the tests passed on this one!?

bugger

### CODE REVIEW

File: lib/spark/dsl/extension.ex [hunk 1]
```diff
@@ -1484,7 +1484,10 @@ defmodule Spark.Dsl.Extension do
                   Spark.CodeHelpers.lift_functions(arg_value, key, __CALLER__)
 
                 if is_nil(arg_value) &&
-                     Enum.any?(entity.args, &(elem(&1, 0) == :optional && elem(&1, 1) == key)) do
+                     Enum.any?(
+                       entity.args,
+                       &(is_tuple(&1) && elem(&1, 0) == :optional && elem(&1, 1) == key)
```
Comment (zachdaniel) [inline lib/spark/dsl/extension.ex:1489]: Should this be `&1 == key || (&(is_tuple(&1) && elem(&1, 0) == :optional && elem(&1, 1) == key))`? Is this only meant to see if there are optional arguments?

Comment (zachdaniel) [reply lib/spark/dsl/extension.ex:1489]: nvm, its clear to me now :)

**Review by zachdaniel — APPROVED** [review]

File: lib/spark/dsl/extension.ex (hunk 1 above)
Comment (jimsynz) [reply lib/spark/dsl/extension.ex:1489]: It's only meant to catch optional arguments named `key`.

Comment (jimsynz) [reply lib/spark/dsl/extension.ex:1489]: It's a bit shit, but I don't want to add yet another function to this module.

**Pull request merged**